include/zephyr/devicetree.h

```diff
@@ -1712,6 +1712,93 @@
  * @}
  */
 
+/**
+ * @defgroup devicetree-generic-vendor Vendor name helpers
+ * @ingroup devicetree
+ * @{
+ */
+
+/**
+ * @brief Get the vendor at index "idx" as a string literal
+ *
+ * The vendor is a string extracted from the vendor prefixes file if an entry
```

Comment:
Docs are wrong due to a subtlety here: you can have as many as one vendor prefixes file per directory in DTS_ROOT.

```
    set(vendor_prefixes ${dts_root}/${VENDOR_PREFIXES})
    if(EXISTS ${vendor_prefixes})
      list(APPEND EXTRA_GEN_DEFINES_ARGS --vendor-prefixes ${vendor_prefixes})
    endif()
  endforeach()
```

(This is how we allow users to define their own to avoid OOT bindings causing spurious warnings.)